content/en/blog/2019/alb-ingress-gateway-iks/index.md

```diff
@@ -0,0 +1,389 @@
+          caCertificates: /etc/istio/ingressgateway-ca-certs/trustid-x3-root.pem
+        hosts:
+        - "*"
+    EOF
+    {{< /text >}}
+
+1. Configure routes for traffic entering via the `Gateway`:
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: VirtualService
+    metadata:
+      name: default-ingress
+    spec:
+      hosts:
+      - "*"
+      gateways:
+      - default-ingress-gateway
+      http:
+      - match:
+        - uri:
+            prefix: /status
+        route:
+        - destination:
+            port:
+              number: 8000
+            host: httpbin.httptools.svc.cluster.local
+    EOF
+    {{< /text >}}
+
+1.  Delete the Istio Ingress Gateway's pod to reload the certificates:
+
+    {{< text bash >}}
+    $ kubectl delete pod -l istio=ingressgateway -n istio-system
+    {{< /text >}}
+
+1.  Send a request to `httpbin` by _curl_, passing as parameters the client certificate
+    (the `--cert` option) and the private key (the `--key` option):
+
+    {{< text bash >}}
+    $ curl https://$INGRESS_GATEWAY_DOMAIN/status/418 --cert alb_certs/client.crt  --key alb_certs/client.key
+
+    -=[ teapot ]=-
+
+       _...._
+     .'  _ _ `.
+    | ."` ^ `". _,
+    \_;`"---"`|//
+      |       ;/
+      \_     _/
+        `"""`
+    {{< /text >}}
+
+1.  Remove the directories with the ALB and ingress gateway certificates and keys.
+
+    {{< text bash >}}
+    $ rm -r ingress_gateway_certs alb_certs trustid-x3-root.pem trusted.crt
+    {{< /text >}}
+
+## Configure ALB
+
+You need to configure your Ingress resource to direct traffic to the Istio Ingress gateway while using the
+certificate stored in the `alb-certs` secret.
+
+1.  Configure the `Ingress` resource for ALB. You must create the `Ingress` resource in the `istio-system` namespace in
+    order to forward the traffic to the Istio ingress gateway.
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    apiVersion: extensions/v1beta1
+    kind: Ingress
+    metadata:
+      name: alb-ingress
+      namespace: istio-system
+      annotations:
+        ingress.bluemix.net/ssl-services: "ssl-service=istio-ingressgateway ssl-secret=alb-certs"
+    spec:
+      tls:
+      - hosts:
+        - httpbin.$ALB_INGRESS_DOMAIN
+        secretName: $ALB_SECRET
+      rules:
+      - host: httpbin.$ALB_INGRESS_DOMAIN
+        http:
+          paths:
+          - path: /status
+            backend:
+              serviceName: istio-ingressgateway
+              servicePort: 443
+    EOF
+    {{< /text >}}
+
+1.  Test the ALB ingress:
+
+    {{< text bash >}}
+    $ curl https://httpbin.$ALB_INGRESS_DOMAIN/status/418
+
+    -=[ teapot ]=-
+
+       _...._
+     .'  _ _ `.
+    | ."` ^ `". _,
+    \_;`"---"`|//
+      |       ;/
+      \_     _/
+        `"""`
+    {{< /text >}}
+
+Congratulations! You configured the IKS Ingress ALB to securely send traffic to Istio ingress gateway.
```

Comment:
I think it would be good to recap and explain how the traffic is mTLS among the IKS ALB to istio ingress gateway.  Basically, for istio ingress gateway, the istio ingress gateway is configured with certs/keys from the ALB domain.  For ALB, user has to append the issuer certificate of [Let's Encrypt](https://letsencrypt.org) to the certificate of ingress gateway so the ingress gateway can validate key/certs from ALB.  Further, from the ALB you direct traffic to istio-ingress gateway at path /status.

Reply:
Done in https://github.com/istio/istio.io/pull/5098/commits/e6307814cc8f1a640f6ad8461994f9573c018f46